File: 06 - Exam Analysis - New/Vcl/other/w2w/1StClass4000/VCL6/cdemos/ButtonGroup/childwin.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "childwin.h"
#include "mainmdi.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "fcDemoRichEdit"
#pragma resource "*.dfm"
TMDIChild *MDIChild;
//---------------------------------------------------------------------------
__fastcall TMDIChild::TMDIChild(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TMDIChild::FormClose(TObject *Sender, TCloseAction &Action)
{
  Action = caFree;        
}
//---------------------------------------------------------------------------

void __fastcall TMDIChild::FormDestroy(TObject *Sender)
{
  MainMdiForm->fcButtonGroup1->ButtonItems->FindPointerTag(this)->Free();
}
//---------------------------------------------------------------------------

void __fastcall TMDIChild::FormActivate(TObject *Sender)
{
 MainMdiForm->fcButtonGroup1->ButtonItems->FindPointerTag(this)->Selected = True;
}
//---------------------------------------------------------------------------

void __fastcall TMDIChild::FormCreate(TObject *Sender)
{
  Graphics::TBitmap *ABitmap;
  TfcButtonGroupItem * ABtnitem;

  ABtnitem = MainMdiForm->fcButtonGroup1->ButtonItems->Add();
  ((TfcShapeBtn *)(ABtnitem->Button))->TextOptions->Alignment = taLeftJustify;
  ABtnitem->PointerTag = this;
  ABitmap = new Graphics::TBitmap;
  ABitmap->Width = 16;
  ABitmap->Height = 16;
  DrawIconEx(ABitmap->Canvas->Handle, 0, 0, Icon->Handle, 16, 16, 0, 0, DI_NORMAL | DI_DEFAULTSIZE);
  ABtnitem->Button->Glyph = ABitmap;
  ABitmap->Free();
}
//---------------------------------------------------------------------------

File: 06 - Exam Analysis - New/Vcl/other/w2w/1StClass4000/VCL6/demos/ButtonGroup/Childwin.pas
unit Childwin;

interface

uses Windows, Classes, Graphics, Forms, Controls, StdCtrls, Main, Buttons, fcCommon, fcShapeBtn,
  ComCtrls, fcDemoRichEdit;

type
  TMDIChild = class(TForm)
    fcDemoRichEdit1: TfcDemoRichEdit;
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure FormCreate(Sender: TObject);
    procedure FormDestroy(Sender: TObject);
    procedure FormActivate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation


{$R *.DFM}

procedure TMDIChild.FormClose(Sender: TObject; var Action: TCloseAction);
begin
  Action := caFree;
end;

procedure TMDIChild.FormCreate(Sender: TObject);
var ABitmap: TBitmap;
begin
  with MainMdiForm.fcButtonGroup1.ButtonItems.Add do
  begin
    (Button as TfcShapeBtn).TextOptions.Alignment := taLeftJustify;
    PointerTag := self;
    ABitmap := TBitmap.Create;
    ABitmap.Width := 16;
    ABitmap.Height := 16;
//    ABitmap.Canvas.Draw(0, 0, Icon);
    DrawIconEx(ABitmap.Canvas.Handle, 0, 0, Icon.Handle, 16, 16, 0, 0, DI_NORMAL or DI_DEFAULTSIZE);
    Button.Glyph := ABitmap;
    ABitmap.Free;
  end;
end;

procedure TMDIChild.FormDestroy(Sender: TObject);
begin
  MainMDIForm.fcButtonGroup1.ButtonItems.FindPointerTag(self).Free;
end;

procedure TMDIChild.FormActivate(Sender: TObject);
begin
  MainMDIForm.fcButtonGroup1.ButtonItems.FindPointerTag(self).Selected := True;
end;

end.
